lib/Controller/MessagesController.php

```diff
@@ -222,6 +222,15 @@ public function getBody(int $accountId, string $folderId, int $messageId): JSONR
 			base64_decode($folderId),
 			$messageId
 		);
+
+		// FIXME: This is probably not a sane way to map the IMAP message to the database
```

Comment:
@ChristophWurst Please have a look at this once you're back. I wasn't able to dig too deep into it to see if there is a better way. Aparently the Message component just gets the data directly from IMAP and therefore the database id that is required for fetching the thread is missing there.

Reply:
Yes, my bad. The restructuring from #3439 fixed this. But good catch :)